aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -715,6 +715,37 @@ Tensor empty_like_sparse_csr(
         self.layout(),
         options.device());
     return result;
+  } else if (options.layout() == kSparseCsc) {
+    auto result = at::native::_sparse_csc_tensor_unsafe(
+        self.ccol_indices().clone(),
```

Comment:
E.g. I think `ccol_indices()` throws on a sparse_csr tensor, etc.

Reply:
You are right, so `layout` will take over `self.layout` in `merge_in`, and we can run into issues when `layout != self.layout` with different compression type.

Reply:
yeah. My impulse is that it's not super important to support the `self.layout != layout` cases unless it's easy, but if we don't, it's probably better UX to throw from here with an explicit message to that effect, rather than letting the called methods blow up. (Would also eliminate the risk of things *not* blowing up and passing garbage to the `..._unsafe` constructors)